test/kubernetes_unittest.cc

```diff
@@ -764,41 +764,69 @@ bool WaitForNewerCollectionTimestamp(const MetadataStore& store,
   return false;
 }
 
-TEST_F(KubernetesTest, KubernetesUpdater) {
+void TestKubernetesUpdater(bool cluster_level_metadata) {
+  const std::string nodes_watch_path = cluster_level_metadata ?
+    "/api/v1/watch/nodes/?watch=true" :
+    "/api/v1/watch/nodes/TestNodeName?watch=true";
+  const std::string pods_watch_path = cluster_level_metadata ?
+    "/api/v1/pods?fieldSelector=spec.nodeName%3D&watch=true" :
+    "/api/v1/pods?fieldSelector=spec.nodeName%3DTestNodeName&watch=true";
+  // These 2 paths only used when cluster_level_metadata is true.
+  const std::string services_watch_path =
+    "/api/v1/watch/services/?watch=true";
+  const std::string endpoints_watch_path =
+    "/api/v1/watch/endpoints/?watch=true";
+
   // Create a fake server representing the Kubernetes master.
   testing::FakeServer server;
   server.SetResponse("/api/v1/nodes?limit=1", "{}");
   server.SetResponse("/api/v1/pods?limit=1", "{}");
-  server.AllowStream(
-      "/api/v1/pods?fieldSelector=spec.nodeName%3DTestNodeName&watch=true");
-  server.AllowStream(
-      "/api/v1/watch/nodes/TestNodeName?watch=true");
+  server.AllowStream(nodes_watch_path);
+  server.AllowStream(pods_watch_path);
+  if (cluster_level_metadata) {
```

Comment:
I wonder if it makes sense to factor out the true common bits, rather than try to conditionalize the implementation of the full test...

Reply:
Done.